src/Storages/FileLog/FileLogSource.cpp

```diff
@@ -56,7 +56,12 @@ FileLogSource::~FileLogSource()
 Chunk FileLogSource::generate()
 {
     if (!buffer || buffer->noRecords())
+    {
+        /// There is no onFinish for ISource, we call it
+        /// when no records return to close files
+        onFinish();
```

Comment:
Then change it for `ISink`? For rabbitmq sink is used.

Reply:
I don't understand, there is no `ISink` used?

Reply:
Sorry, a little different:
https://github.com/ClickHouse/ClickHouse/blob/dc657a2c5eb57862e9387e62bd6548863ff1db87/src/Storages/RabbitMQ/RabbitMQSink.h#L10